web-server/src/components/Settings/SyncDaysSetting.tsx

```diff
@@ -0,0 +1,137 @@
+import { FlexBox } from '../FlexBox';
+import { Box, Divider, TextField } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import { Line } from '../Text';
+import DoneIcon from '@mui/icons-material/Done';
+import { FC, useEffect, useState } from 'react';
+import { handleApi } from '@/api-helpers/axios-api-instance';
+import { useAuth } from '@/hooks/useAuth';
+import { OrgResetBookmarkApiResponse } from '@/types/resources';
+import { useBoolState } from '@/hooks/useEasyState';
+import { useSnackbar } from 'notistack';
+import { useDispatch, useSelector } from '@/store';
+import {
+  getDefaultSyncDaysSettings,
+  updateDefaultSyncDaysSettings
+} from '@/slices/org';
+
+export const SyncDaysSetting: FC = () => {
+  const dispatch = useDispatch();
+  const inputFocus = useBoolState(false);
+  const isSaving = useBoolState(false);
+  const [syncDays, setSyncDays] = useState<number>(31);
+  const { orgId } = useAuth();
+  const { enqueueSnackbar } = useSnackbar();
+  const initialSyncDays = useSelector((state) => state.org.defaultSyncDays);
+
+  const getSyncDays = async (): Promise<number> => {
+    const res = await dispatch(getDefaultSyncDaysSettings({ orgId }));
+    return res.payload as number;
+  };
+
+  const updateSyncDays = async (days: number): Promise<number> => {
+    const res = await dispatch(
+      updateDefaultSyncDaysSettings({ orgId, defaultSyncDays: days })
+    );
+    return res.payload as number;
+  };
+
+  const handleSave = async () => {
+    isSaving.true();
+    const fallbackSyncDays = initialSyncDays;
+    try {
+      await updateSyncDays(syncDays);
+      handleApi<OrgResetBookmarkApiResponse>(`/internal/${orgId}/bookmarks`, {
+        method: 'PUT'
+      })
+        .then(() => {
+          enqueueSnackbar('Default Sync Days Updated Successfully', {
+            variant: 'success',
+            autoHideDuration: 3000
+          });
+        })
+        .catch(() => {
+          updateSyncDays(fallbackSyncDays);
+          setSyncDays(fallbackSyncDays);
+        });
```

Comment:
I think all this should be part of the same thunk that updates sync days.
Since both of those actions needs to succeed collectively.

In any case, @samad-yar-khan believes it's based to not make the UI aware of this technical detail, and the backend should internally handle it. So this would change as soon as he's done with his change.

We'll have to wait till then.

Reply:
handled in #518